File: app/src/main/java/com/qimian233/ztool/settingactivity/setting/magicwindowsearch/PackageInfo.java
// PackageInfo.java
package com.qimian233.ztool.settingactivity.setting.magicwindowsearch;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class PackageInfo {
    private String name;
    private String mainPage;
    private List<ActivityPair> activityPairs;
    private List<String> forceFullscreenPages;
    private List<String> transActivities;
    private List<String> leftTransActivities;
    private String showEmbeddingDivider;
    private String skipLetterboxDisplayInfo;
    private String skipMultiWindowMode;
    private String showSurfaceViewBackground;
    private String shouldPausePrimaryActivity;

    public PackageInfo(JSONObject jsonObject) throws JSONException {
        this.name = jsonObject.optString("name", "");
        this.mainPage = jsonObject.optString("mainPage", "");

        // 解析activityPairs
        this.activityPairs = new ArrayList<>();
        if (jsonObject.has("activityPairs")) {
            JSONArray pairsArray = jsonObject.getJSONArray("activityPairs");
            for (int i = 0; i < pairsArray.length(); i++) {
                JSONObject pairObj = pairsArray.getJSONObject(i);
                activityPairs.add(new ActivityPair(
                        pairObj.optString("from", ""),
                        pairObj.optString("to", "")
                ));
            }
        }

        // 解析forceFullscreenPages
        this.forceFullscreenPages = new ArrayList<>();
        if (jsonObject.has("forceFullscreenPages")) {
            JSONArray fullscreenArray = jsonObject.getJSONArray("forceFullscreenPages");
            for (int i = 0; i < fullscreenArray.length(); i++) {
                forceFullscreenPages.add(fullscreenArray.getString(i));
            }
        }

        // 解析transActivities
        this.transActivities = new ArrayList<>();
        if (jsonObject.has("transActivities")) {
            JSONArray transArray = jsonObject.getJSONArray("transActivities");
            for (int i = 0; i < transArray.length(); i++) {
                transActivities.add(transArray.getString(i));
            }
        }

        // 解析leftTransActivities
        this.leftTransActivities = new ArrayList<>();
        if (jsonObject.has("leftTransActivities")) {
            JSONArray leftTransArray = jsonObject.getJSONArray("leftTransActivities");
            for (int i = 0; i < leftTransArray.length(); i++) {
                leftTransActivities.add(leftTransArray.getString(i));
            }
        }

        // 解析其他字符串字段
        this.showEmbeddingDivider = jsonObject.optString("showEmbeddingDivider", "未设置");
        this.skipLetterboxDisplayInfo = jsonObject.optString("skipLetterboxDisplayInfo", "未设置");
        this.skipMultiWindowMode = jsonObject.optString("skipMultiWindowMode", "未设置");
        this.showSurfaceViewBackground = jsonObject.optString("showSurfaceViewBackground", "未设置");
        this.shouldPausePrimaryActivity = jsonObject.optString("shouldPausePrimaryActivity", "未设置");
    }

    // Getter方法
    public String getName() { return name; }
    public String getMainPage() { return mainPage; }
    public List<ActivityPair> getActivityPairs() { return activityPairs; }
    public List<String> getForceFullscreenPages() { return forceFullscreenPages; }
    public List<String> getTransActivities() { return transActivities; }
    public List<String> getLeftTransActivities() { return leftTransActivities; }
    public String getShowEmbeddingDivider() { return showEmbeddingDivider; }
    public String getSkipLetterboxDisplayInfo() { return skipLetterboxDisplayInfo; }
    public String getSkipMultiWindowMode() { return skipMultiWindowMode; }
    public String getShowSurfaceViewBackground() { return showSurfaceViewBackground; }
    public String getShouldPausePrimaryActivity() { return shouldPausePrimaryActivity; }
}

File: app/src/main/java/com/qimian233/ztool/screens/zuisetting/magicwindowsearch/PackageInfo.kt
package com.qimian233.ztool.screens.zuisetting.magicwindowsearch

import org.json.JSONException
import org.json.JSONObject

class PackageInfo @Throws(JSONException::class) constructor(jsonObject: JSONObject) {
    val name: String = jsonObject.optString("name", "")
    val mainPage: String = jsonObject.optString("mainPage", "")

    // Parse activityPairs
    val activityPairs: List<ActivityPair> = buildList {
        if (jsonObject.has("activityPairs")) {
            val pairsArray = jsonObject.getJSONArray("activityPairs")
            for (i in 0 until pairsArray.length()) {
                val pairObj = pairsArray.getJSONObject(i)
                add(ActivityPair(pairObj.optString("from", ""), pairObj.optString("to", "")))
            }
        }
    }

    // Parse forceFullscreenPages
    val forceFullscreenPages: List<String> = buildList {
        if (jsonObject.has("forceFullscreenPages")) {
            val fullscreenArray = jsonObject.getJSONArray("forceFullscreenPages")
            for (i in 0 until fullscreenArray.length()) {
                add(fullscreenArray.getString(i))
            }
        }
    }

    // Parse transActivities
    val transActivities: List<String> = buildList {
        if (jsonObject.has("transActivities")) {
            val transArray = jsonObject.getJSONArray("transActivities")
            for (i in 0 until transArray.length()) {
                add(transArray.getString(i))
            }
        }
    }

    // Parse leftTransActivities
    val leftTransActivities: List<String> = buildList {
        if (jsonObject.has("leftTransActivities")) {
            val leftTransArray = jsonObject.getJSONArray("leftTransActivities")
            for (i in 0 until leftTransArray.length()) {
                add(leftTransArray.getString(i))
            }
        }
    }

    // Parse other string fields
    val showEmbeddingDivider: String = jsonObject.optString("showEmbeddingDivider", "未设置")
    val skipLetterboxDisplayInfo: String = jsonObject.optString("skipLetterboxDisplayInfo", "未设置")
    val skipMultiWindowMode: String = jsonObject.optString("skipMultiWindowMode", "未设置")
    val showSurfaceViewBackground: String = jsonObject.optString("showSurfaceViewBackground", "未设置")
    val shouldPausePrimaryActivity: String = jsonObject.optString("shouldPausePrimaryActivity", "未设置")
}
